testing/functional_tests_assertions.rst

```diff
@@ -0,0 +1,79 @@
+.. index::
+   single: Tests; Assertions
+
+Functional Test specific assertions Reference
+=============================================
+
+When doing functional tests, sometimes you need to make complex assertions in
+order to check whether the ``Request``, the ``Response`` or the ``Crawler``
+contain the expected information to make your test succeed.
+
+Here is an example without the specific assertions::
+
+    $this->assertGreaterThan(
+        0,
+        $crawler->filter('html:contains("Hello World")')->count()
+    );
+
+Now here is the example with the assertions introduced in Symfony 4.3::
+
+    $this->assertSelectorTextContains('html', 'Hello World');
+
+As you can see, even though the first assertion uses native PHPUnit methods,
+the second one explains more what we want to expect.
+
+As of Symfony 4.3, a list of assertions have been added, here is the reference.
+
+.. note::
+
+    These assertions will not work with `symfony/panther`_ as they use the
+    ``Request`` and ``Response`` objects coming from the ``HttpFoundation``
+    component, and the ``KernelBrowser`` coming from the ``FrameworkBundle``.
+    Panther only uses the ``BrowserKit` component.
+
+.. note::
+
+    Most of these assertions only work if a request has been made with the ``Client``
+    in a test case extending the ``WebTestCase`` class.
+
+Response assertions
+~~~~~~~~~~~~~~~~~~~
+
+- ``assertResponseIsSuccessful(string $message = '')``
```

Comment:
To be less verbose in the docs and be safe if changing/adding some method signature I propose to use:
```suggestion
- :method:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase::assertResponseIsSuccessful`
```

Reply:
These links do not work anymore as we have remove the Symfony API docs. They now redirect to Github (but not on the exact line where the method is defined).

Reply:
Ok, didn't now that, thanks for the information. But to me sth like
```suggestion
- ``assertResponseIsSuccessful()``
```
is much more readable and less noisy

Reply:
I agree, we don't need all the method arguments here. IDEs are here to help with that.

Reply:
Fixed, thanks 🙂